bundles/org.openhab.binding.samsungtv/src/main/java/org/openhab/binding/samsungtv/internal/SamsungTvTlsTrustManagerProvider.java

```diff
@@ -0,0 +1,82 @@
+/**
+ * Copyright (c) 2010-2019 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.samsungtv.internal;
+
+import java.net.Socket;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+
+import javax.net.ssl.SSLEngine;
+import javax.net.ssl.X509ExtendedTrustManager;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.smarthome.io.net.http.TlsTrustManagerProvider;
+import org.osgi.service.component.annotations.Component;
+
+/**
+ * Provides a TrustManager to allow secure websocket connections to any TV (=server)
+ *
+ * @author Arjan Mels - Initial Contribution
+ */
+@Component
+@NonNullByDefault
+public class SamsungTvTlsTrustManagerProvider implements TlsTrustManagerProvider {
+    @Override
+    public String getHostName() {
+        return "SmartViewSDK";
+    }
+
+    private final X509ExtendedTrustManager trustAllCerts = new X509ExtendedTrustManager() {
+
+        @Override
+        public void checkClientTrusted(X509Certificate @Nullable [] x509Certificates, @Nullable String s)
+                throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate @Nullable [] x509Certificates, @Nullable String s)
+                throws CertificateException {
+        }
+
+        @Override
+        public X509Certificate @Nullable [] getAcceptedIssuers() {
+            return null;
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate @Nullable [] x509Certificates, @Nullable String s,
+                @Nullable Socket socket) throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate @Nullable [] x509Certificates, @Nullable String s,
+                @Nullable Socket socket) throws CertificateException {
+        }
+
+        @Override
+        public void checkClientTrusted(X509Certificate @Nullable [] x509Certificates, @Nullable String s,
+                @Nullable SSLEngine sslEngine) throws CertificateException {
+        }
+
+        @Override
+        public void checkServerTrusted(X509Certificate @Nullable [] x509Certificates, @Nullable String s,
+                @Nullable SSLEngine sslEngine) throws CertificateException {
+        }
+    };
+
+    @Override
+    public X509ExtendedTrustManager getTrustManager() {
+        return trustAllCerts;

```

Comment:
@arjanmels I was a little bit of the radar, but great that you succeeded in using it like this 👍 

A possible improvement would be to actually use the [TrustAllTrustMananger](https://github.com/openhab/openhab-core/blob/master/bundles/org.openhab.core.io.net/src/main/java/org/eclipse/smarthome/io/net/http/TrustAllTrustMananger.java)

Reply:
Hi, Thanks for getting back on this and for the suggestion. I will take it on board for a next version.

Is there also an option to allow all certificates for all hosts? I am not certain all SamsungTV's use "SmartViewSDK" as CN. I also experimented with 127.0.0.1 as that seems to be picked up as alternative host name. What do you think?

Reply:
Please keep away from `127.0.0.1` because that would impact other bindings using the HttpClient or WebsocketClient, the current solution would hopefully be enough for now :-)

Reply:
Ok, The TrustAllTrustManager is a good idea, but not yet available in OH2.4, which makes testing for a wider group more difficult, so at the moment I will stick to the custom implementation.